src/Symfony/Component/Form/Extension/Core/Type/ImmutableChoiceType.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\ChoiceList\Loader\ChoiceLoaderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Base class for child of {@link ChoiceType}.
+ *
+ * Every class extending it is responsible for loading
+ * the choices and eventually cache them.
+ *
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+abstract class ImmutableChoiceType extends AbstractType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults(array(
+            'choice_loader' => $this->getChoiceLoader(),
+            'choice_translation_domain' => false,
+        ));
+
+        $resolver->setRequired(array('choice_loader'));
```

Comment:
I guess it's better to be strict in an abstract class here and `getChoiceLoader` as abstract looks safer to maintain.